pkg/utils/cmd/cmdutil.go

```diff
@@ -0,0 +1,220 @@
+	if err != nil {
+		return false, fmt.Errorf("%s: %w", flagName, err)
+	}
+
+	if str == "" {
+		return defaultValue, nil
+	}
+
+	value, err := strconv.ParseBool(str)
+	if err != nil {
+		return false, fmt.Errorf("invalid value for %s [%s]: %w", flagName, str, err)
+	}
+
+	return value, nil
+}
+
+// GetDuration returns values either command line flag or environment variable.
+func GetDuration(cmd *cobra.Command, flagName, envKey string,
+	defaultDuration time.Duration, isOptional bool) (time.Duration, error) {
+	timeoutStr, err := GetUserSetVarFromString(cmd, flagName, envKey, isOptional)
+	if err != nil {
+		return -1, err
+	}
+
+	if timeoutStr == "" {
+		return defaultDuration, nil
+	}
+
+	timeout, err := time.ParseDuration(timeoutStr)
+	if err != nil {
+		return -1, fmt.Errorf("invalid value [%s]: %w", timeoutStr, err)
+	}
+
+	return timeout, nil
+}
+
+// GetInt returns values either command line flag or environment variable.
+func GetInt(cmd *cobra.Command, flagName, envKey string, defaultValue int, isOptional bool) (int, error) {
+	str, err := GetUserSetVarFromString(cmd, flagName, envKey, isOptional)
+	if err != nil {
+		return 0, fmt.Errorf("%s: %w", flagName, err)
+	}
+
+	if str == "" {
+		return defaultValue, nil
+	}
+
+	value, err := strconv.Atoi(str)
+	if err != nil {
+		return 0, fmt.Errorf("invalid value for %s [%s]: %w", flagName, str, err)
+	}
+
+	return value, nil
+}
+
+// GetFloat returns values either command line flag or environment variable.
+func GetFloat(cmd *cobra.Command, flagName, envKey string, defaultValue float64, isOptional bool) (float64, error) {
+	str, err := GetUserSetVarFromString(cmd, flagName, envKey, isOptional)
+	if err != nil {
+		return 0, fmt.Errorf("%s: %w", flagName, err)
+	}
+
+	if str == "" {
+		return defaultValue, nil
+	}
+
+	value, err := strconv.ParseFloat(str, 64)
+	if err != nil {
+		return 0, fmt.Errorf("invalid value for %s [%s]: %w", flagName, str, err)
+	}
+
+	return value, nil
+}
+
+// TLSParameters contains TLS parameters retrieved from command arguments.
+type TLSParameters struct {
+	SystemCertPool bool
+	CACerts        []string
+	ServeCertPath  string
+	ServeKeyPath   string
+}
+
+// TLSFields contains TLS command field names and environment variables key.
+type TLSFields struct {
+	SystemCertPoolFlagName string
+	SystemCertPoolEnvKey   string
+	CACertsFlagName        string
+	CACertsEnvKey          string
+	CertificateFlagName    string
+	CertificateLEnvKey     string
+	KeyFlagName            string
+	KeyEnvKey              string
+}
+
+// GetTLS returns values either command line flag or environment variable.
+func GetTLS(cmd *cobra.Command, tlsFields *TLSFields) (*TLSParameters, error) {
+	tlsSystemCertPoolString := GetUserSetOptionalVarFromString(cmd, tlsFields.SystemCertPoolFlagName,
+		tlsFields.SystemCertPoolEnvKey)
+
+	tlsSystemCertPool := false
+
+	if tlsSystemCertPoolString != "" {
+		var err error
+
+		tlsSystemCertPool, err = strconv.ParseBool(tlsSystemCertPoolString)
```

Comment:
You can use GetBool instead.

Reply:
I guess parsing the actual value is better than calling GetBool and fetch the value from the arg flagName/envKey again